core/resourcemanager.js

```diff
@@ -136,14 +136,21 @@ CKEDITOR.resourceManager.prototype = {
 	 * @param {String} names The resource names, separated by commas.
 	 * @param {String} path The path of the folder containing the resource.
 	 * @param {String} [fileName] The resource file name. If not provided, the
-	 * default name is used. If provided with a empty string, will implicitly indicates that `path` argument
+	 * default name is used. If provided with an empty string, will implicitly indicates that `path` argument
 	 * is already the full path.
 	 */
 	addExternal: function( names, path, fileName ) {
 		names = names.split( ',' );
 		for ( var i = 0; i < names.length; i++ ) {
```

Comment:
I know that it's out of the scope of this issue, but it seems that the [logic for setting correct path and filename](https://github.com/ckeditor/ckeditor-dev/blob/4033248f9e6fead2b7ffc9d2b973e67ec309be34/core/resourcemanager.js#L147-L154) can be moved outside the loop. The whole function can look something like:

```javascript
addExternal: function( names, path, fileName ) {
	names = names.split( ',' );

	// If there is no '/' at the end of path, but 'fileName' is anything different
	// than an empty string (so either is not set at all or is a valid name), assume
	// that path is a directory and concatenate '/' to it (#917).
	if ( path.charAt( path.length - 1 ) != '/' && fileName !== '' ) {
		path += '/';
	}

	// If "fileName" is not provided, we assume that it may be available
	// in "path". Try to extract it in this case.
	if ( !fileName ) {
		path = path.replace( /[^\/]+$/, function( match ) {
			fileName = match;
			return '';
		} );
	}

	// Use the default file name if there is no "fileName" and it
	// was not found in "path".
	fileName = fileName || ( this.fileName + '.js' );

	CKEDITOR.tools.array.forEach( names, function( name ) {
		this.externals[ name ] = {
			dir: path,
			file: fileName
		};
	} );
}
```

As we look inside this file rarely, let's fix this small issue now and just forget about it :D

Reply:
We won't do any unnecessary operations several times, inside the loop; just once before it. And the code will become more logical.

Reply:
Ok, I meant that we never use functionality for loading multiple files at once, but I understand it's JIC. Thanks for tip.